### PR TITLE
Ajna actions2

### PR DESCRIPTION
**Pull request by paszkowskiDamian**

### CODE REVIEW

File: contracts/actions/ajna/DepositBorrow.sol [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+   * @param data Encoded calldata that conforms to the BorrowData struct
+   */
+  function execute(bytes calldata data, uint8[] memory) external payable override {
+    DepositBorrowData memory args = parseInputs(data);
+    IAjnaPool pool = IAjnaPool(args.pool);
+    IAjnaPoolUtilsInfo poolUtilsInfo = IAjnaPoolUtilsInfo(registry.getRegisteredService(AJNA_POOL_UTILS_INFO));
+
+    uint256 index = poolUtilsInfo.priceToIndex(args.price);
+
+    pool.drawDebt(address(this), args.borrowAmount * pool.quoteTokenScale(), index, args.borrowAmount * pool.collateralScale());
```
Comment (halaprix) [inline contracts/actions/ajna/DepositBorrow.sol:31]: `args.borrowAmount` is used twice

File: contracts/actions/ajna/RepayWithdraw.sol [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+   */
+  function execute(bytes calldata data, uint8[] memory) external payable override {
+    RepayWithdrawData memory args = parseInputs(data);
+    IAjnaPool pool = IAjnaPool(args.pool);
+    IAjnaPoolUtilsInfo poolUtilsInfo = IAjnaPoolUtilsInfo(registry.getRegisteredService(AJNA_POOL_UTILS_INFO));
+
+    uint256 index = poolUtilsInfo.priceToIndex(args.price);
+
+    pool.repayDebt(address(this), args.repayAmount * pool.quoteTokenScale(), args.withdrawAmount * pool.collateralScale(), address(this), index);
+    store().write(bytes32(args.repayAmount));
```
Comment (halaprix) [inline contracts/actions/ajna/RepayWithdraw.sol:32]: the order is different thatn in `DepositBorrow` is it intended?

- 1st: debt 
- 2nd: collateral

**Review by halaprix — CHANGES_REQUESTED** [review]
two comments

File: contracts/actions/ajna/RepayWithdraw.sol (hunk 2 above)
Comment (paszkowskiDamian) [reply contracts/actions/ajna/RepayWithdraw.sol:32]: it was supposed to reflect the order of actual operations, first you need to repay debt and you can withdraw, also this is exactly the same as in action name.

**Comment by paszkowskiDamian** [comment]
@halaprix https://github.com/OasisDEX/oasis-earn-sc/pull/280 requested changes were applied here.

**Pull request closed without merging**